simpl/tests/test_external_schema.py

```diff
@@ -0,0 +1,349 @@
+        instance = baker.make(Instance, run=self.run, _fill_optional=True)
+        result = schema.execute(
+            """
+            query ($runId: ID!) {
+                run(id: $runId) {
+                    instances {
+                        name
+                        status
+                        dateEnd
+                        players
+                        playerCount
+                    }
+                }
+            }""",
+            variable_values={"runId": self.run.id},
+            context_value=self.context,
+        )
+
+        self.assertEqual(
+            result.data,
+            {
+                "run": {
+                    "instances": [
+                        {
+                            "dateEnd": instance.date_end.isoformat(),
+                            "name": instance.name,
+                            "status": instance.status.name,
+                            "players": [],
+                            "playerCount": 0,
+                        },
+                    ],
+                }
+            },
+        )
+
+    def test_get_instance_with_multiple_players(self):
+        instance = baker.make(Instance, run=self.run)
+
+        # Create users with Auth0 IDs
+        user1 = baker.make(SocialAccount, provider="auth0", uid="auth0_1")
+        user2 = baker.make(SocialAccount, provider="auth0", uid="auth0_2")
+
+        # Create a user without an Auth0 ID
+        user3 = baker.make(SocialAccount, provider="google", uid="google")
+
+        # Create a user without SocialAccount
+        user4 = baker.make(User)
+
+        # Create characters linking the users to the instance
+        baker.make("simpl.Character", instance=instance, user=user1.user)
+        baker.make("simpl.Character", instance=instance, user=user2.user)
+        baker.make("simpl.Character", instance=instance, user=user3.user)
+        baker.make("simpl.Character", instance=instance, user=user4)
+
+        result = schema.execute(
+            """
+            query ($runId: ID!) {
+                run(id: $runId) {
+                    instances {
+                        players
+                        playerCount
+                    }
+                }
+            }""",
+            variable_values={"runId": self.run.id},
+            context_value=self.context,
+        )
+
+        instance_data = result.data["run"]["instances"][0]
+
+        # There should be 4 players
+        self.assertEqual(instance_data["playerCount"], 4)
+        # Users with Auth0 IDs should be in the list of players
+        self.assertIn(user1.uid, instance_data["players"])
+        self.assertIn(user2.uid, instance_data["players"])
+        # Users without Auth0 IDs should not be in the list of players
+        self.assertNotIn(user3.uid, instance_data["players"])
+        # Users without SocialAccounts should not be in the list of players
+        self.assertNotIn(user4.id, instance_data["players"])
```

Comment:
Here we are basically writing a new test that asserts behaviour that we've decided is a bug.

i.e: We're writing a test that says `playerCount` should be 4 and there should only be 3 users in `players`.

That's kind of fine, given we've decided to leave it as-is for now, not get bogged down in that yet, and make going down that specific rabbit hole another ticket. That said lets note it in the comments round this, just in case that follow-up ticket doesn't get followed up immediately.

Reply:
You're right, I'm gonna clarify that a bit :)